pkg/controllers/management/auth/legacy_rt_cleaner.go

```diff
@@ -28,6 +28,11 @@ func (p *rtCleaner) sync(key string, obj *v3.RoleTemplate) (runtime.Object, erro
 		return obj, nil
 	}
 
+	return p.mgmt.Management.RoleTemplates("").Update(obj)
+}
+
+// rtCleanUp returns a clean RoleTemplate based on filters specified within the function
+func rtCleanUp(obj *v3.RoleTemplate) *v3.RoleTemplate {
```

Comment:
I think you're missing a call to `rtCleanUp` in the `sync` function. I'm guessing that's what is causing the integration test failure too, because it's looking for the annotation to match but we never actually clean up the annotations.

Reply:
you might be right, let me modify this and see what happen

Reply:
This worked! Thanks @JonCrowther!!